lifecycle-operator/controllers/common/eventsender.go

```diff
@@ -15,28 +20,93 @@ type IEvent interface {
 
 // ===== Main =====
 
-func NewEventSender(recorder record.EventRecorder) IEvent {
-	return newK8sSender(recorder)
+type EventMultiplexer struct {
+	logger   logr.Logger
+	emitters []IEvent
+}
+
+func NewEventMultiplexer(logger logr.Logger, recorder record.EventRecorder, client ce.Client) *EventMultiplexer {
+	multiplexer := &EventMultiplexer{
+		logger: logger,
+	}
+	multiplexer.register(newCloudEventSender(logger, client))
+	multiplexer.register(NewK8sSender(recorder))
+	return multiplexer
+}
+
+func (e *EventMultiplexer) register(emitter IEvent) {
+	if emitter != nil {
+		e.emitters = append(e.emitters, emitter)
+	}
+}
+
+func (e *EventMultiplexer) Emit(phase apicommon.KeptnPhaseType, eventType string, reconcileObject client.Object, status string, message string, version string) {
+	for _, emitter := range e.emitters {
+		e.logger.Info(fmt.Sprintf("Emitting event using %T", emitter))
+		emitter.Emit(phase, eventType, reconcileObject, status, message, version)
+	}
 }
 
 // ===== Cloud Event Sender =====
-// TODO: implement Cloud Event logic
+
+type cloudEvent struct {
+	client ce.Client
+	logger logr.Logger
+}
+
+func newCloudEventSender(logger logr.Logger, client ce.Client) *cloudEvent {
+	return &cloudEvent{
+		client: client,
+		logger: logger,
+	}
+}
+
+// Emit creates a Cloud Event and send it to the endpoint
+func (e *cloudEvent) Emit(phase apicommon.KeptnPhaseType, eventType string, reconcileObject client.Object, status string, message string, version string) {
+	endpoint := config.Instance().GetCloudEventsEndpoint()
+	if endpoint == "" {
+		// if no endpoint is configured we don't emit any event
+		if !strings.HasPrefix(endpoint, "http") {
+			e.logger.V(5).Info(fmt.Sprintf("CloudEvent endpoint configured but it does not start with http: %s", endpoint))
+		}
+		return
+	}
+	event := ce.NewEvent()
+	event.SetSource("keptn.sh")
+	event.SetType(fmt.Sprintf("%s.%s", phase.LongName, status))
```

Comment:
I think the event.Time property might be set automatically by the CloudEvent client when it's sent, but can you double check if that's the case? If not, let's set the time here

Reply:
it is indeed added by default. I will add an assert to verify that the property is set